lib/daru/dataframe.rb

```diff
@@ -2157,41 +2157,53 @@ def access_vector *names
       end
     end
 
-    def access_row *names
-      location = names[0]
+    def access_row *args
+      # location = names[0]
 
-      if @index.is_a?(MultiIndex)
-        pos = @index[names]
-        if pos.is_a?(Integer)
-          return Daru::Vector.new(populate_row_for(pos), index: @vectors, name: pos)
-        end
+      # if @index.is_a?(MultiIndex)
+      #   pos = @index[names]
+      #   if pos.is_a?(Integer)
+      #     return Daru::Vector.new(populate_row_for(pos), index: @vectors, name: pos)
+      #   end
 
-        new_rows = pos.map { |tuple| populate_row_for(tuple) }
+      #   new_rows = pos.map { |tuple| populate_row_for(tuple) }
 
-        if !location.is_a?(Range) && names.size < @index.width
-          pos = pos.drop_left_level names.size
-        end
+      #   if !location.is_a?(Range) && names.size < @index.width
+      #     pos = pos.drop_left_level names.size
+      #   end
 
-        Daru::DataFrame.rows(new_rows, order: @vectors, name: @name, index: pos)
-      else
-        if names[1].nil?
-          names = @index[location]
-          if names.is_a?(Numeric)
-            row = []
-            @data.each do |vector|
-              row << vector[location]
-            end
+      #   Daru::DataFrame.rows(new_rows, order: @vectors, name: @name, index: pos)
+      # else
+      #   if names[1].nil?
+      #     names = @index[location]
+      #     if names.is_a?(Numeric)
+      #       row = []
+      #       @data.each do |vector|
+      #         row << vector[location]
+      #       end
+
+      #       return Daru::Vector.new(row, index: @vectors, name: set_name(location))
+      #     end
+      #   end
+      #   # Access multiple rows
+      #   rows = []
+      #   names.each do |name|
+      #     rows << self.row[name].to_a
+      #   end
 
-            return Daru::Vector.new(row, index: @vectors, name: set_name(location))
-          end
-        end
-        # Access multiple rows
-        rows = []
-        names.each do |name|
-          rows << self.row[name].to_a
-        end
+      #   Daru::DataFrame.rows rows, index: names,name: @name, order: @vectors
+      # end
+      
+      return nil if args.first.is_a? Range
+      positions = @index.pos(*args)
 
-        Daru::DataFrame.rows rows, index: names,name: @name, order: @vectors
+      if positions.is_a? Numeric
+        return Daru::Vector.new populate_row_for(positions), index: @vectors
+      else
+        new_rows = @data.map { |vec| vec[*args] }
+        p new_rows
+        return Daru::DataFrame.new new_rows, index: @index.subset(*args),
+          order: @vectors
```

Comment:
Column-wise operations will take up considerable time since you'll to first create a Daru::Vector and then give it to the user. Also then you'll need some clever architectural changes to make the benefits of Vectors available to DataFrame (like it's APIs).

Also you'll need to make massive non-trivial changes throughout the library in order to reflect these changes. If there is a better way to store data, it should be first properly researched by looking at other similar libraries and how they store data internally.